src/Interpreters/Context.cpp

```diff
@@ -1523,17 +1530,40 @@ void Context::resetZooKeeper() const
     shared->zookeeper.reset();
 }
 
+static void reloadZooKeeperIfChangedImpl(const ConfigurationPtr & config, const std::string & config_name, zkutil::ZooKeeperPtr & zk)
+{
+    if (!zk || zk->configChanged(*config, config_name))
+        zk = std::make_shared<zkutil::ZooKeeper>(*config, config_name);
+    if (zk->expired())
```

Comment:
But how we can get into this branch? Seems like this logic is not consistent with function name?

Reply:
Still don't understand why we need it.